### DISCUSSION TITLE
v3.3.0 - New theme and improved usability

### DISCUSSION THREAD

**Discussion by Jacalz** [Jacalz/rymdport#57] [Announcements]
This release focuses on improving the usability and design. The most notable change is that Rymdport now will check for new updates on startup and then ask the user if it should automatically update and restart the application. On top of this there is a new theme, a few minor design design tweaks and support for following FreeDesktop Dark Style Preference on Linux/BSD.

That is just the tip of the iceberg. Read the full changelog below to see all of the exciting changes :)

### Full changelog
- The application can now automatically update itself.
- Notifications for received files now mentions where the file was saved to.
- Present more useful information about how to write custom codes.
- The dialog for custom codes now focuses the text entry automatically.
- The selection of the download path has been reworked to be more useful.
- Minor redesign of the send and receive tabs.
- Minor wording improvements inside the settings tab.
- Updated `fyne` to [v2.3.0](https://github.com/fyne-io/fyne/releases/tag/v2.3.0).
  - An entirely new theme with improved visuals.
  - The application will now follow the FreeDesktop Dark Style Preference on Linux/BSD.
  - Added an option to create a new folder within the folder selection dialog.
  - Lots of other improvements and fixes.
- Update `compress` to [v1.15.14](https://github.com/klauspost/compress/releases/tag/v1.15.14).
  - Includes performance improvements that benefit sending and receiving of directories.
- Minimal supported Go compiler version is now Go 1.17.
- Release binaries are now built using Go 1.19.
  - Various performance improvements, security fixes and other improvements.

### A big thanks to Fyne Labs
The automatic application updates would not have been possible without the help of [Fyne Labs](https://fynelabs.com/). They have kindly provided access to their automated cross-compilation platform [Geoffrey](https://fynelabs.com/geoffrey/) as well as infrastructure and hosting for the automatic updates. I am very grateful for their support and cooperation.

### Note to packagers

The automatic updating is disabled by default as it does not fit in well where package managers handle updates.
Automatic updates are only enabled when passing `-tags release` to the Go compiler only and is of course enabled for all of the release binaries here on GitHub.

<hr /><em>This discussion was created from the release <a href='https://github.com/Jacalz/rymdport/releases/tag/v3.3.0'>v3.3.0 - New theme and improved usability</a>.</em>

**Comment by aniqueta** [comment]
My Apple Silicon machine running macOS Ventura says the [rymdport-v3.3.0-macOS-arm64](https://github.com/Jacalz/rymdport/releases/download/v3.3.0/rymdport-v3.3.0-macOS-arm64.zip) app is damaged.

**Reply by Jacalz** [reply]
Hmm. I believe Apple treats the binaries as such because they are not signed by a trusted certificate. There should be some setting that allows you to lower that setting, I think.

**Reply by Jacalz** [reply]
I don't know the exact setting on the top of my head but the amd64 binary should work fine through Rosetta 2 in the meantime

**Reply by aniqueta** [reply]
Running `xattr -r -d com.apple.quarantine <path to .app>` resolves the issue with the arm64 build. Thanks.

**Reply by Jacalz** [reply]
That's great. Thanks for sharing. Will add that to the documentation :)

**Reply by Jacalz** [reply]
I added a section to the README about that command in https://github.com/Jacalz/rymdport/commit/b6979ac486088521d600168efc7946d40c7b8a8a. Thanks for the help solving that issue :)
